File: esports-backend/src/main/java/com/esports/entity/Wallet.java
package com.esports.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "wallets")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Wallet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "available_balance", nullable = false)
    private Double availableBalance = 0.0;

    @Column(name = "escrow_balance", nullable = false)
    private Double escrowBalance = 0.0;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

    // Null-safe balance operations
    public void addToAvailableBalance(Double amount) {
        if (amount != null && amount > 0) {
            this.availableBalance = (this.availableBalance != null ? this.availableBalance : 0.0) + amount;
        }
    }

    public void subtractFromAvailableBalance(Double amount) {
        if (amount != null && amount > 0) {
            this.availableBalance = Math.max(0.0, (this.availableBalance != null ? this.availableBalance : 0.0) - amount);
        }
    }

    public void addToEscrowBalance(Double amount) {
        if (amount != null && amount > 0) {
            this.escrowBalance = (this.escrowBalance != null ? this.escrowBalance : 0.0) + amount;
        }
    }

    public void subtractFromEscrowBalance(Double amount) {
        if (amount != null && amount > 0) {
            this.escrowBalance = Math.max(0.0, (this.escrowBalance != null ? this.escrowBalance : 0.0) - amount);
        }
    }
}

File: Upkart1/Entities/Wallet.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Upkart1.Entities
{
    [Table("wallets")]
    public class Wallet
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("user_id")]
        public long UserId { get; set; }
        public User User { get; set; } = null!;

        [Required]
        [Column("available_balance")]
        public double AvailableBalance { get; set; } = 0.0;

        [Required]
        [Column("escrow_balance")]
        public double EscrowBalance { get; set; } = 0.0;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<WalletTransaction> WalletTransactions { get; set; } = new List<WalletTransaction>();

        // Null-safe balance operations matching Spring backend
        public void AddToAvailableBalance(double? amount)
        {
            if (amount != null && amount > 0)
            {
                AvailableBalance += amount.Value;
            }
        }

        public void SubtractFromAvailableBalance(double? amount)
        {
            if (amount != null && amount > 0)
            {
                AvailableBalance = Math.Max(0.0, AvailableBalance - amount.Value);
            }
        }

        public void AddToEscrowBalance(double? amount)
        {
            if (amount != null && amount > 0)
            {
                EscrowBalance += amount.Value;
            }
        }

        public void SubtractFromEscrowBalance(double? amount)
        {
            if (amount != null && amount > 0)
            {
                EscrowBalance = Math.Max(0.0, EscrowBalance - amount.Value);
            }
        }
    }
}
